src/tests/Interop/COM/NETClients/Primitives/Program.cs

```diff
@@ -20,22 +20,52 @@ static int Main(string[] doNotUse)
 
             try
             {
+                Console.WriteLine(
+@"=========================
+Testing COM object interop.
+=========================");
                 RunTests();
-                Console.WriteLine("Testing COM object lifetime control methods.");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Test object interop failure: {e}");
+                return 101;
+            }
+
+            try
+            {
+                Console.WriteLine(
+@"
+=========================
+Testing COM object lifetime control methods.
+=========================");
                 Thread.CurrentThread.DisableComObjectEagerCleanup();
                 RunTests();
+
+                // Force GC to ensure all RCWs have been considered and are ready to be cleaned up.
+                ForceGC();
                 Marshal.CleanupUnusedObjectsInCurrentContext();
                 Assert.IsFalse(Marshal.AreComObjectsAvailableForCleanup());
             }
             catch (Exception e)
             {
-                Console.WriteLine($"Test Failure: {e}");
+                Console.WriteLine($"Test lifetime control failure: {e}");
                 return 101;
             }
 
             return 100;
         }
 
+        private static void ForceGC()
+        {
+            GC.Collect();
+            GC.Collect();
```

Comment:
What are the redundant `GC.Collect` calls meant to collect?

If the redundant `GC.Collect` calls are meant to collect stuff released by the finalizers, it may be better to do it like this:
```
for (int i = 0; i < 3; i++)
{
    GC.Collect();
    GC.WaitForPendingFinalizers();
}
```

Reply:
This is how I was instructed to called `GC.Collect()` long ago. Is the correct way what you have written?

Reply:
It depends on what you are trying to achieve.

Reply:
Sure. The multiple calls are to ensure the GC has considered as much as possible. I think your proposal is clearer as to my intent though.